### ISSUE TITLE
Use gamma-cat to create all-sky exclusion mask

### ISSUE THREAD

**Issue by adonath** [gammapy/gammapy#943]
We have a function [make_tevcat_exclusion_mask](https://github.com/gammapy/gammapy/blob/master/gammapy/image/mask.py#L201), which uses TeVCat to create an all-sky exclusion mask. This should be changed to use [gamma-cat](https://github.com/gammapy/gamma-cat).

In addition we should think about storing the mask as a high resolution healpix FITS image to avoid artefacts cause by the projection. Downside is that user have to install `healpy`....

**Comment by cdeil** [comment]
I've removed `make_tevcat_exclusion_mask` in #1009 .

This is what I think we should do to clear this out:
- [ ] Add a small utility function to make an exclusion mask from a table with RA, DEC, Radius (similar to the old `make_tevcat_exclusion_mask`, but not hardwired to TeVCat.
- [ ] In examples, use `gamma-cat` instead. Yes, also make an all-sky mask and put it in gammapy-extra as @adonath suggests
- [ ] Clean up the remaining traces of tevcat (use gamma-cat or a simpler example instead) in `gammapy` and `gammapy-extra` (see below)

Putting this under the 0.7 milestone. 

```
$ ack tevcat

examples/test_image_analysis.py
66:    exclusion_mask = SkyImage.read('$GAMMAPY_EXTRA/datasets/exclusion_masks/tevcat_exclusion.fits')

gammapy/background/tests/test_reflected.py
15:    filename = '$GAMMAPY_EXTRA/datasets/exclusion_masks/tevcat_exclusion.fits'

gammapy/cube/tests/test_cube_pipe.py
93:    exclusion_mask = SkyImage.read('$GAMMAPY_EXTRA/datasets/exclusion_masks/tevcat_exclusion.fits')

gammapy/data/tests/test_obs_stats.py
39:    return SkyImage.read('$GAMMAPY_EXTRA/datasets/exclusion_masks/tevcat_exclusion.fits')

gammapy/data/tests/test_obs_summary.py
61:        exclusion_mask = SkyImage.read('$GAMMAPY_EXTRA/datasets/exclusion_masks/tevcat_exclusion.fits')

gammapy/scripts/cube_background.py
135:    catalog = load_catalog_tevcat()

gammapy/scripts/tests/test_image_pipe.py
58:    exclusion_mask = SkyImage.read('$GAMMAPY_EXTRA/datasets/exclusion_masks/tevcat_exclusion.fits')

gammapy/time/tests/test_lightcurve.py
66:    exclusion_file = '$GAMMAPY_EXTRA/datasets/exclusion_masks/tevcat_exclusion.fits'
```
In gammapy-extra:
```
$ find . -name '*tevcat*'
./datasets/catalogs/make_tevcat.py
./datasets/catalogs/tevcat.fits.gz
./datasets/catalogs/tevcat_extra.fits.gz
./datasets/exclusion_masks/make_tevcat_exclusion_mask
./datasets/exclusion_masks/tevcat_exclusion.fits
```

**Comment by cdeil** [comment]
This is mostly done, I removed the TeVCat catalog just now:
https://github.com/gammapy/gammapy-extra/commit/7b34e6748d5982ac4581912dbe77949380ef6de7

What remains to be done is to adapt those tests to not use this file:
```
gammapy/background/tests/test_reflected.py
15:    filename = '$GAMMAPY_EXTRA/datasets/exclusion_masks/tevcat_exclusion.fits'

gammapy/time/tests/test_lightcurve.py
162:    exclusion_file = '$GAMMAPY_EXTRA/datasets/exclusion_masks/tevcat_exclusion.fits'

gammapy/data/tests/test_obs_stats.py
30:    return SkyImage.read('$GAMMAPY_EXTRA/datasets/exclusion_masks/tevcat_exclusion.fits')

gammapy/data/tests/test_obs_summary.py
61:        exclusion_mask = SkyImage.read('$GAMMAPY_EXTRA/datasets/exclusion_masks/tevcat_exclusion.fits')
```
and then finally to delete  `$GAMMAPY_EXTRA/datasets/exclusion_masks/tevcat_exclusion.fits`

I will do this after #1391 is done, then it should be easier because the mask is no longer needed in the tests.

Using an all-sky exclusion mask was never a good idea.

**Comment by cdeil** [comment]
@registerrier - Assigning to you.

**Comment by registerrier** [comment]
I suppose we still need a utility function to make a mask for given `MapGeom` from `gamma-cat` ?

**Comment by cdeil** [comment]
@registerrier removed the users of the tevcat mask in #1613 and I removed the file from gammapy-extra just now in  https://github.com/gammapy/gammapy-extra/commit/e9c678b7e801f3a094eeb5cbb8a6c8ad8c991084

Closing this issue.

Note that there's still #1612 open to add a docs example for exclusion mask, and #1555 to update tutorial notebooks in the next 1-2 days before the v0.8 release.
